xbmc/utils/SaveFileStateJob.cpp

```diff
@@ -183,23 +182,32 @@ void CSaveFileState::DoWork(CFileItem& item,
           if (!videodatabase.GetStreamDetails(dbItem) ||
               dbItem.GetVideoInfoTag()->m_streamDetails != item.GetVideoInfoTag()->m_streamDetails)
           {
-            const int idFile = videodatabase.SetStreamDetailsForFile(
+            videoDbSuccess = videodatabase.SetStreamDetailsForFile(
                 item.GetVideoInfoTag()->m_streamDetails, progressTrackingFile);
+            updateListing = !videoDbSuccess;
+          }
+        }
 
-            if (idFile == -2)
+        // See if fileId needs updating
+        if (URIUtils::IsBluray(item.GetDynPath()))
+        {
+          if (item.GetVideoContentType() == VideoDbContentType::MOVIES)
+          {
+            const CVideoInfoTag* tag{item.GetVideoInfoTag()};
+            if (tag->m_strFileNameAndPath != item.GetDynPath())
             {
-              videoDbSuccess = false;
+              const int currentFileId{videodatabase.GetFileIdByMovie(tag->m_iDbId)};
+              videoDbSuccess |= videodatabase.SetFileForMovie(item.GetDynPath(), currentFileId);
```

Comment:
I don't think that's going to work for versions that are not the default version of a movie. If tag->m_iFileId points to the originally played file, then it could be used and eliminate the need for GetFileIdByMovie at the same time.
For safety, passing the iDbId and the current file id to SetFileForMovie would be good.